### ISSUE TITLE
Misleading CLI exit codes

### ISSUE THREAD

**Issue by jaley** [Azure/aztk#604]
We've had some automated jobs silently failing due to exit code 0 being returned in these two error conditions:

* Cluster provisioning with `aztk spark cluster create` failed due to Batch account out of quota
* Subsequent attempt to submit a job to that cluster with `aztk spark cluster submit` failed because no master has been selected (or cluster not ready for whatever reason)

Would it be possible to make sure any failed CLI command returns a non-zero exit status in error conditions such as these?

**Comment by timotheeguerin** [comment]
Yeah it seems like when there is an error we just log it but end successfully. We should update that.

**Comment by jafreck** [comment]
My guess is there are more cases where this happens than noted above. We should try to get full coverage here or this will keep coming back.

**Comment by ukadiri** [comment]
We are facing similar issues even with 0.10.2 is there any workaround for these issues?
